lua/nvim-treesitter/query_predicates.lua

```diff
@@ -85,5 +89,32 @@ query.add_predicate('has-type?', function(match, pattern, bufnr, pred)
   return vim.tbl_contains(types, node:type())
 end)
 
+local function str_match(match, pattern, bufnr, pred)
+  if not valid_args(pred[1], pred, 2) then return end
+
+  local case_insensitive = pred[1]:find('case-insensitive', 1, true)
+
+  local fun
+  if case_insensitive then
+      fun = string.lower
+  else
+      fun = noop
+  end
+
+  local ts_utils = require'nvim-treesitter.ts_utils'
+  local node = match[pred[2]]
+  local node_text = fun(ts_utils.get_node_text(node, bufnr)[1])
+
+  local dict = {}
+  for i=3,#pred do
+      dict[fun(pred[i])] = true
```

Comment:
why do you need to create a dictionary when you can compare directly to match? Lua string comparisons are only pointer comparisons since all strings are interned.

Reply:
Ops. Originally I was trying to do something else and left this as mistake. I will fix it.

Reply:
I think `any-of` is a better name. Also, prefixing with `case-insensitive`  makes `not-case-insensitive-foobar` semantically ambiguous.